net-libs/libssh/libssh-0.8.4.recipe

```diff
@@ -57,43 +58,63 @@ BUILD_PREREQUIRES="
 	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
 	cmd:ld$secondaryArchSuffix
-	cmd:libtoolize
 	cmd:make
 	cmd:pkg_config$secondaryArchSuffix
 	"
 
+if [ -z "$secondaryArchSuffix" ]; then
+	ARCHITECTURES_doc="any"
+
+	PROVIDES_doc="
+		libssh_doc = $portVersion
+		"
+
+	BUILD_PREREQUIRES+="
+		cmd:doxygen
+		"
+fi
+
 defineDebugInfoPackage libssh$secondaryArchSuffix \
 	"$libDir"/libssh.so.$libVersion
 
 PATCH()
 {
-	# keyfile torture test is currently broken
+	# Requires: https://review.haiku-os.org/c/haiku/+/632
 	sed -i \
 		-e '/torture_threads_pki_rsa/d' \
 		-e '/torture_rand/d' \
 		-e '/torture_threads_init/d' \
 		-e '/torture_threads_buffer/d' \
 		-e '/torture_threads_crypto/d' \
-		tests/unittests/CMakeLists.txt || die
+		tests/unittests/CMakeLists.txt
+
+	# Doesn't work in chroot, probably due to the lack of the passwd database
+	sed -i 's/cmocka_unit_test(torture_path_expand_tilde_unix),//' \
+		tests/unittests/torture_misc.c
 }
 
 BUILD()
 {
+	_maybe_pie=()
+	if [ "$targetArchitecture" != x86_gcc2 ]; then
```

Comment:
Check my 3 last commits :P

Bash is a much more horrible beast than I expected. I've tested this on gcc2 actually, and seems to work fine.

Reply:
My previous suggestion to check `[ $effectiveTargetArchitecture" != x86_gcc2 ]` was wrong because the recipe does not build for x86_gcc2 primary arch.
So the right test is neither `[ "$targetArchitecture" != x86_gcc2 ]` nor `[ "$effectiveTargetArchitecture" != x86_gcc2 ]` but  `[ "$effectiveTargetArchitecture" = x86_64 ]`.
(OK, `[ "$targetArchitecture" != x86_gcc2 ]` works for x86_gcc2 hybrid and x86_64, but it would not work for x86 primary arch, so what we need is really `[ "$effectiveTargetArchitecture" = x86_64 ]`.)

Reply:
`100% tests passed, 0 tests failed out of 20` on x86 second arch and x86_64.
I'll test it asap on x86 primary arch to see if we need this change:
```diff
  	_maybe_pie=()
-	if [ "$targetArchitecture" != x86_gcc2 ]; then
+	if [ "$effectiveTargetArchitecture" = x86_64 ]; then
 		_maybe_pie=("-DCMAKE_EXE_LINKER_FLAGS=-pie -fstack-protector")
 	fi
```

Reply:
Hmm, I don't have gcc7 on my x86 primary arch, and libssh is failing to build with gcc 5.4.0.
Anyway, the above can't be wrong. I'll push it for you, then I'll let you squash and merge with your GPG signature :)